src/api/File.js

```diff
@@ -81,6 +85,51 @@ class File extends Item {
         this.successHandler(`${downloadBaseUrl}?${downloadUrlQuery}`);
     }
 
+    /**
+     * Polls a representation's infoUrl, attempting to generate a representation
+     *
+     * @param {FileRepresentation} representation - representation that should have it's info.url polled
+     * @return {Promise<FileRepresentation>} - representation updated with most current status
+     */
+    async generateRepresentation(representation: FileRepresentation): Promise<FileRepresentation> {
+        const infoUrl = getProp(representation, 'info.url');
+
+        const numOfTries = 4;
+        const initialTimeout = 2000;
+        const backoffFactor = 2;
+
+        if (!infoUrl) {
+            return representation;
+        }
+
+        const isStatusSuccess = response => {
+            const status = getProp(response, 'data.status.state');
+            return (
+                !status ||
+                status === REPRESENTATIONS_RESPONSE_ERROR ||
+                status === REPRESENTATIONS_RESPONSE_SUCCESS ||
+                status === REPRESENTATIONS_RESPONSE_VIEWABLE
+            );
+        };
+
+        return this.xhr.get({ url: infoUrl }).then(() =>
```

Comment:
Does this first call need to be duplicated or can we return `retryNumOfTimes` directly?